test/e2e/upgrade_test.go

```diff
@@ -282,3 +331,98 @@ func (tc *testContext) waitForCSINodesWithDrivers(nodes []v1.Node) error {
 		return true, nil
 	})
 }
+
+// deployUnhealthyNodesCheckerJob deploys a Job that checks the number of nodes marked as unhealthy and fails if the
+// number of unhealthy nodes exceeds a threshold
```

Comment:
Why does this have be done with a Pod?

Reply:
I don't understand your explanation. You should be able to make the API calls directly from the Pod from where we are running the tests without having to launch a Job on the test cluster. That by definition is external. This is similar to [testNodesBecomeReadyAndSchedulable()](https://github.com/openshift/windows-machine-config-operator/blob/d775f625191d16730f8e86d42f98a9ebbe921753/test/e2e/validation_test.go#L59C27-L59C27). If you want to run something in the background, you can use a Go subroutine and channels.

Reply:
> I don't understand your explanation

I can set up a meeting for more details. LMK

WRT running in the background with  Go subroutine and channels take into account there is no subroutine and main thread in this case. The `vsphere-e2e-upgrade` job  a) [setup the upgrade](https://github.com/openshift/windows-machine-config-operator/blob/master/hack/run-ci-e2e-test.sh#L176), b) [triggers the upgrade](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml#L177) and c) [tests the upgrade](https://github.com/openshift/windows-machine-config-operator/blob/master/hack/run-ci-e2e-test.sh#L186C4-L186C4),  where the steps are scattered between the release repo and wmco e2e test suite.

Reply:
Yes, please set up a meeting

Reply:
Updated https://github.com/openshift/windows-machine-config-operator/pull/1901/commits/830c233d53aef2fe256581deec0949fb1b569ef0 with details in the commit message.